doc/code/qml_pulse.rst

```diff
@@ -0,0 +1,215 @@
+    ops = [XX, YY, ZZ]
+    H2 =  qml.dot(coeffs, ops)
+
+
+.. warning::
+    When initializing a :class:`~.ParametrizedHamiltonian` via a list of parametrized coefficients, it
+    is possible to create a list of multiple coefficients of the same form iteratively using lambda
+    functions, i.e.
+
+    ``coeffs = [lambda p, t: p for _ in range(3)]``.
+
+    Do **not**, however, define the function as dependent on the value that is iterated over. That is, it is not
+    possible to define ``coeffs = [lambda p, t: p * t**i for i in range(3)]`` to create a list
+    ``coeffs = [(lambda p, t: p), (lambda p, t: p * t), (lambda p, t: p * t**2)]``. The value of ``i`` when
+    creating the lambda functions is set to be the final value in the iteration, such that this will
+    produce three identical functions ``coeffs = [(lambda p, t: p * t**2)] * 3``.
+
+The :class:`~.ParametrizedHamiltonian` is a callable, and can return an :class:`~.Operator` if passed a set of
+parameters and a time at which to evaluate the coefficients :math:`f_j`.
+
+.. code-block:: python
+
+    >>> H1
+    ParametrizedHamiltonian: terms=3
+
+    >>> params = [1.2, [2.3, 3.4]]  # f1 takes a single parameter, f2 takes 2
+    >>> H1(params, t=0.5)
+    (2*(PauliX(wires=[0]) @ PauliX(wires=[1]))) + ((-0.2876553535461426*(PauliY(wires=[0]) @ PauliY(wires=[1]))) + (1.5179612636566162*(PauliZ(wires=[0]) @ PauliZ(wires=[1]))))
+
+
+When passing parameters, ensure that the order of the coefficient functions and the order of
+the parameters match.
+
+When initializing a :class:`~.ParametrizedHamiltonian`, terms defined with fixed coefficients have to come
+before parametrized terms to prevent discrepancy in the wire order.
+
+
+ParametrizedEvolution
+---------------------
+During a pulse sequence spanning time :math:`(t_0, t_1)`, the state evolves according to the time-dependent Schrodinger equation
+
+.. math:: \frac{\partial}{\partial t} |\psi\rangle = -i H(t) |\psi\rangle
+
+realizing a unitary evolution :math:`U(t_0, t_1)` of the input state, i.e.
+
+.. math:: |\psi(t_1)\rangle = U(t_0, t_1) |\psi(t_0)\rangle
+
+A :class:`~.ParametrizedEvolution` is this solution :math:`U(t_0, t_1)` to the time-dependent
+Schrodinger equation for a :class:`~.ParametrizedHamiltonian`.
+
+.. note::
+    The :class:`~.ParametrizedHamiltonian` must be Hermitian at all times. This is not explicitly
+    checked; ensuring a correctly defined Hamiltonian is the responsibility of the user.
+
+The :class:`~.ParametrizedEvolution` class uses a numerical ordinary differential equation
+solver (see `jax.experimental.ode <https://github.com/google/jax/blob/main/jax/experimental/ode.py>`_). It
+can be created using the :func:`~.functions.evolve` function:
+
+.. code-block:: python
+
+    from jax import numpy as jnp
+
+    f1 = lambda p, t: p * jnp.sin(t) * (t - 1)
+    H = 2 * qml.PauliX(0) + f1 * qml.PauliY(1)
+    ev = qml.evolve(H)
+    ev
+    >>> ParametrizedEvolution(wires=[0, 1])
```

Comment:
Always separate code blocks and `pycon` commands, otherwise the `copy` option in the docs doesn't work.

```suggestion
.. code-block:: python

    from jax import numpy as jnp

    f1 = lambda p, t: p * jnp.sin(t) * (t - 1)
    H = 2 * qml.PauliX(0) + f1 * qml.PauliY(1)
    ev = qml.evolve(H)
    ev
    
>>> ParametrizedEvolution(wires=[0, 1])
```